lib/ansible/modules/system/vdo.py

```diff
@@ -0,0 +1,762 @@
+              The 'sparse' mode has a deduplication window of 1 GB of
+              index memory per 10 TB of incoming data, but requires
+              100 GB of index data on persistent storage.
+        required: false
+    ackthreads:
+        description:
+            - Specifies the number of threads to use for
+              acknowledging completion of requested VDO I/O operations.
+              Valid values are integer values from 1 to 100 (lower
+              numbers are preferable due to overhead).  The default is
+              1.
+        required: false
+    biothreads:
+        description:
+            - Specifies the number of threads to use for submitting I/O
+              operations to the storage device.  Valid values are
+              integer values from 1 to 100 (lower numbers are
+              preferable due to overhead).  The default is 4.
+        required: false
+    cputhreads:
+        description:
+            - Specifies the number of threads to use for CPU-intensive
+              work such as hashing or compression.  Valid values are
+              integer values from 1 to 100 (lower numbers are
+              preferable due to overhead).  The default is 2.
+        required: false
+    logicalthreads:
+        description:
+            - Specifies the number of threads across which to
+              subdivide parts of the VDO processing based on logical
+              block addresses.  Valid values are integer values from
+              1 to 100 (lower numbers are preferable due to overhead).
+              The default is 1.
+        required: false
+    physicalthreads:
+        description:
+            - Specifies the number of threads across which to
+              subdivide parts of the VDO processing based on physical
+              block addresses.  Valid values are integer values from
+              1 to 16 (lower numbers are preferable due to overhead).
+              The physical space used by the VDO volume must be
+              larger than (slabsize * physicalthreads).  The default
+              is 1.
+        required: false
+notes:
+  - In general, the default thread configuration should be used.
+'''
+
+EXAMPLES = '''
+# Create a VDO volume
+- name: Create 2 TB VDO volume vdo1 on device /dev/md0
+  vdo:
+    name: vdo1
+    state: present
+    device: /dev/md0
+    logicalsize: 2T
+
+# Remove a VDO volume
+- name: Remove VDO volume vdo1
+  vdo:
+    name: vdo1
+    state: absent
+'''
+
+RETURN = '''# '''
+
+from ansible.module_utils.basic import AnsibleModule
+import re
+
+try:
+    import yaml
+    HAS_YAML = True
+except ImportError:
+    HAS_YAML = False
+
+
+# Generate a list of VDO volumes, whether they are running or stopped.
+#
+# @param module  The AnsibleModule object.
+# @param vdo_cmd  The path of the 'vdo' command.
+#
+# @return vdolist  A list of currently created VDO volumes.
+def inventory_vdos(module, vdo_cmd):
```

Comment:
not a blocker: you could just make module and vdo_cmd globals, this is a short running script in it's own process, so globals are not that bad